File: app/src/main/java/com/study/fldemo/DefineApplication.java
package com.study.fldemo;

import android.app.Application;
import android.content.Context;

import com.study.fldemo.utils.SpUtils;


import androidx.multidex.MultiDex;

/**
 * Created by zouqianyu on 2017/10/1.
 */

public class DefineApplication extends Application {
    public static boolean loginState = false;
    public static SpUtils spUtils;

    @Override
    public void onCreate() {
        super.onCreate();
        spUtils = SpUtils.getInstance(getApplicationContext());
    }

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        MultiDex.install(this);
    }


}

File: app/src/main/java/com/study/fldemo/DefineApplication.kt
package com.study.fldemo

import android.app.Application
import android.content.Context

import com.study.fldemo.utils.SpUtils


import androidx.multidex.MultiDex

/**
 *
 */

class DefineApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        spUtils = SpUtils.getInstance(applicationContext)
        defineApplicationContext = applicationContext
    }

    override fun attachBaseContext(base: Context) {
        super.attachBaseContext(base)
        MultiDex.install(this)
    }

    companion object {
        var loginState = false
        lateinit var spUtils: SpUtils
        lateinit var defineApplicationContext: Context

        fun getContext(): Context {
            return defineApplicationContext
        }
    }


}
